src/mono/mono.proj

```diff
@@ -14,29 +14,21 @@
   <PropertyGroup>
     <MonoCrossDir Condition="'$(MonoCrossDir)' == '' and '$(ROOTFS_DIR)' != ''">$(ROOTFS_DIR)</MonoCrossDir>
     <MonoForceInterpreter Condition="'$(MonoForceInterpreter)' == ''">false</MonoForceInterpreter>
-    <ExeExt Condition="'$(OS)' == 'Windows_NT'">.exe</ExeExt>
     <ScriptExt Condition="'$(OS)' == 'Windows_NT'">.cmd</ScriptExt>
     <ScriptExt Condition="'$(OS)' != 'Windows_NT'">.sh</ScriptExt>
-    <ExeExt Condition="'$(OS)' == 'Windows_NT'">.exe</ExeExt>
-    <LibPrefix Condition="'$(TargetsWindows)' != 'true'">lib</LibPrefix>
-    <SharedLibExt Condition="'$(TargetsOSX)' == 'true' or '$(TargetsMacCatalyst)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">.dylib</SharedLibExt>
-    <SharedLibExt Condition="'$(TargetsWindows)' == 'true'">.dll</SharedLibExt>
-    <SharedLibExt Condition="'$(SharedLibExt)' == ''">.so</SharedLibExt>
-    <StaticLibExt Condition="'$(TargetsWindows)' == 'true'">.lib</StaticLibExt>
-    <StaticLibExt Condition="'$(StaticLibExt)' == ''">.a</StaticLibExt>
```

Comment:
Is `LibSuffix` going to be the right thing for our cross-compilation targets? (ie will this play nice with src/mono/monoaotcross.proj ?) /cc @directhex

Reply:
The conditions in naming/props are based on `PackageRID`, which undergoes complicated calculation which accounts for cross, portable and other aspects of matrix: https://github.com/dotnet/runtime/blob/ca072ba73c92e5730b382806291e6d1b48d7fbfd/Directory.Build.props#L158
Lets fix it here if something is missing (we will find out by the CI run).

Reply:
@directhex Please make sure to validate the official build to make sure we don't have any cross-compilation packaging issues.

Reply:
I've scheduled an official build here: https://dev.azure.com/dnceng/internal/_build/results?buildId=1512127&view=results

Reply:
The official build is green so we should be good here :)